frontend/src/components/ProductCarousel.vue

```diff
@@ -9,58 +9,55 @@
-                    :style="{ transform: `translateX(-${(currentIndex * 100) / 5}%)` }">
-                    <!-- Loop through trending products and display them in the carousel -->
-                    <router-link :to="{ name: 'product', params: { product_ref: product.ref } }"
-                        v-for="product in productsOnTrending" :key="product.id"
-                        class="flex-none w-full sm:w-1/2 lg:w-1/5 px-2 cursor-pointer">
-                        <div class="shadow">
-                            <img :src="product.gallery_urls[0]" :alt="product.product_detail.name_en"
-                                class="w-full object-cover mb-4" />
-                            <div class="text-center">
-                                <h3 class="text-lg font-semibold">
-                                    <!-- Display product name based on current language -->
-                                    <span v-if="currentLanguage === 'en'">{{ product.product_detail.name_en }}</span>
-                                    <span v-else>{{ product.product_detail.name_es }}</span>
-                                </h3>
-                                <p class="text-lg font-semibold">
-                                    ${{ product.product_detail.price }}
-                                </p>
-                            </div>
-                        </div>
-                    </router-link>
-                </ul>
-            </div>
-            <!-- Next button for carousel -->
-            <button
-                class="absolute right-0 top-1/2 transform -translate-y-1/2 bg-black text-white rounded-full shadow size-6 flex items-center justify-center z-10"
-                @click="next">
-                <span class="sr-only">Next</span>
-                <span class="block text-sm leading-none">&rsaquo;</span>
-            </button>
+            <Swiper 
+            ref="swiper"
+            :modules="modules" 
+            :loop="true"
+            :breakpoints="{
+                640: {
+                    slidesPerView: 1,
+                    spaceBetween: 20,
+                },
+                768: {
+                    slidesPerView: 2,
+                    spaceBetween: 40,
+                },
+                1024: {
+                    slidesPerView: 4,
+                    spaceBetween: 50,
+                },
+            }" 
+            :autoplay="{
+                delay: 4000,
+                disableOnInteraction: false,
+            }" 
+            class="w-full h-full"
+            >
+                <swiper-slide v-for="product in productsOnTrending">
+                    <img :src="product.gallery_urls[0]">
+                    <h2 v-if="currentLanguage === 'en'" class="text-center text-xl font-medium mt-4">{{ product.product_detail.name_en }}</h2>
+                    <h2 v-else class="text-center text-xl font-medium mt-4">{{ product.product_detail.name_es }}</h2>
```

Comment:
I have noticed that for all the translations we are creating two tags and we are repeating the classes, to avoid this there should be a single tag with its classes and within the parent tag, there should be a <span> tag where the condition is added.

Could you please clean this up, identify all the translations and apply this refactor

Reply:
Defined as follow-up task